athena-jdbc/src/main/java/com/amazonaws/athena/connectors/jdbc/manager/JdbcRecordHandler.java

```diff
@@ -163,7 +163,12 @@ public void readWithConstraint(BlockSpiller blockSpiller, ReadRecordsRequest rea
                 }
                 LOGGER.info("{} rows returned by database.", rowsReturnedFromDatabase);
 
-                connection.commit();
+                // SqlServer jdbc driver is using @@TRANCOUNT while performing commit(), it results below RuntimeException.
```

Comment:
instead of putting this detail in the superclass, just override `readWithConstraint` to not `commit()`. Also, it's unclear to me that just removing `commit()` is enough -- do we start the transaction, and then simply never end it? Should we remove transactions entirely for this connector? all the more reason to override this function.

Reply:
Will do override in sub class.
We are not removing the transactions for this connector. Rather avoiding execution of connection.commit() only in case of Azure Serverless sql environment.